### PR TITLE
[Formatter] Fix formatting issues in several parser nodes

### PR DESCRIPTION
**Pull request by IrushiL**
## Purpose
Fixes formatting issues in the following nodes.
* Do statement.
* Query expressions.
* Fork statements.
* Panic statements.
* Continue statements.
* Break statements.
* Enum declarations.
* Compound assignment statements.
* Let expression.
* Call statements.
* If-else statements.

## Check List 
- [X] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by SupunS — APPROVED** [review]

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormatterUtils.java [hunk 1]
```diff
@@ -526,8 +561,12 @@ private static Indentation getBlockParent(Node node) {
         if (parent.kind() == SyntaxKind.IF_ELSE_STATEMENT) {
             ArrayList nestedBlock = nestedIfBlock((NonTerminalNode) parent);
             if (!nestedBlock.isEmpty()) {
+                boolean addSpaces = false;
+                if (parent.parent() != null && parent.parent().kind() == SyntaxKind.BLOCK_STATEMENT) {
```
Comment (nadeeshaan) [inline misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormatterUtils.java:565]: Shall we assign the addSpaces value with the if's condition? No need to add an if clause just for setting the variable

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NodeIndentation.java [hunk 2]
```diff
@@ -111,8 +112,20 @@ static int simpleNameReferenceNode(Node node, FormattingOptions options) {
             }
         }
         if (parent != null && parent.kind() == SyntaxKind.FUNCTION_CALL) {
+            if (grandParent != null && grandParent.kind() == SyntaxKind.CALL_STATEMENT) {
```
Comment (nadeeshaan) [inline misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NodeIndentation.java:115]: can simplify the addSpaces value assignment by assigning the if condition directly

**Review by nadeeshaan — CHANGES_REQUESTED** [review]

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java [hunk 3]
```diff
@@ -409,12 +418,18 @@ public ReturnTypeDescriptorNode transform(ReturnTypeDescriptorNode returnTypeDes
         if (!isInLineRange(returnTypeDescriptorNode, lineRange)) {
             return returnTypeDescriptorNode;
         }
+        boolean addSpaces = false;
+        if (returnTypeDescriptorNode.parent() != null &&
+                returnTypeDescriptorNode.parent().kind() == SyntaxKind.BLOCK_STATEMENT) {
+            addSpaces = true;
+        }
```
Comment (nadeeshaan) [inline misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java:425]: Can simplify the addSpaces variable assignment

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormatterUtils.java (hunk 1 above)
Comment (IrushiL) [reply misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormatterUtils.java:565]: Will make the following improvement in a refactoring PR.
`boolean addSpaces = parent.parent() != null && parent.parent().kind() == SyntaxKind.BLOCK_STATEMENT;`

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NodeIndentation.java (hunk 2 above)
Comment (IrushiL) [reply misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NodeIndentation.java:115]: Will make the following improvement in a refactoring PR.
`addSpaces = grandParent != null && grandParent.kind() == SyntaxKind.CALL_STATEMENT;`

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormatterUtils.java (hunk 1 above)
Comment (nadeeshaan) [reply misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormatterUtils.java:565]: Great, approving the PR upon the above comment

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java (hunk 3 above)
Comment (IrushiL) [reply misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java:425]: Will make the following improvement in a refactoring PR.
`boolean addSpaces = returnTypeDescriptorNode.parent() != null &&
                returnTypeDescriptorNode.parent().kind() == SyntaxKind.BLOCK_STATEMENT;`

**Review by nadeeshaan — APPROVED** [review]

**Pull request merged**